### ISSUE TITLE
FillPrefix on K32W platform ends up writing way out of bounds of the buffer

### ISSUE THREAD

**Issue by bzbarsky-apple** [project-chip/connectedhomeip#4729]
#### Problem
`chip::Logging::kMaxPrefixLen` is currently set to 3 (was 6 before https://github.com/project-chip/connectedhomeip/pull/4627/).  This value is only used in `src/platform/K32W/Logging.cpp`, in code that looks like this:
```
    VerifyOrDie(bufLen > chip::Logging::kMaxPrefixLen);
    ::GetMessageString(buf, chipCategory, otLevelLog);

    prefixLen = strlen(buf);
```
Note that `GetMessageString` does NOT get the buffer length.  Then `GetMessageString` puts one of the following strings into `buf`: `"[Error]"`, `"[Progress]"`, `"[Debug]"`, `"[Warn]"`, `"[Info]"`.  It does NOT put in the null-terminator for any of those; just the chars before that.

Problems:

1) None of these strings are small enough to fit in `kMaxPrefixLen`.  And even before #4627 they were not small enough to include their null-terminator in there.
2) That `strlen` call will return some random number, because there's no null-terminator being placed into the buffer.

 #### Proposed Solution
Stop writing buffers out of bounds.

**Comment by bzbarsky-apple** [comment]
Implementation note for this and https://github.com/project-chip/connectedhomeip/issues/4732: It might be good to add a utility constexpr variadic template somewhere in the CHIP "support" stuff that takes as args literal strings and returns the max of their lengths.  We could then use that to easily compute the max length we need to have for the prefix bits here.

**Comment by bzbarsky-apple** [comment]
> @bzbarsky-apple was this fixed by #4732?

Different platform.  This is about K32W, not EFR32.  

That said, the relevant code is now in `src/platform/nxp/k32w/k32w0/Logging.cpp` and no longer uses kMaxPrefixLen.

In fact, it looks like the rewrite from #8451 incidentally fixed this.
